module-system/module-implementations/sov-evm/src/evm/transaction.rs

```diff
@@ -29,11 +31,51 @@ impl Default for BlockEnv {
 #[cfg_attr(
     feature = "native",
     derive(serde::Serialize),
-    derive(serde::Deserialize),
-    derive(schemars::JsonSchema)
+    derive(serde::Deserialize)
 )]
 #[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq, Clone)]
 pub struct RlpEvmTransaction {
     /// Rlp data.
     pub rlp: Vec<u8>,
 }
+
+#[cfg_attr(
+    feature = "native",
+    derive(serde::Serialize),
+    derive(serde::Deserialize)
+)]
+#[derive(Debug, PartialEq, Clone)]
+pub struct TransactionSignedAndRecovered {
+    /// Signer of the transaction
+    pub signer: Address,
+    /// Signed transaction
+    pub signed_transaction: TransactionSigned,
+    /// Block the transaction was added to
+    pub block_number: u64,
+}
+
+#[cfg_attr(
+    feature = "native",
+    derive(serde::Serialize),
+    derive(serde::Deserialize)
+)]
+#[derive(Debug, PartialEq, Clone)]
+pub struct Block {
+    /// Block header.
+    pub header: SealedHeader,
+
+    /// Transactions in this block.
+    pub transactions: Range<u64>,
+}
+
+#[cfg_attr(
+    feature = "native",
+    derive(serde::Serialize),
+    derive(serde::Deserialize)
+)]
+#[derive(Debug, PartialEq, Clone)]
+pub struct Receipt {
+    pub receipt: reth_primitives::Receipt,
```

Comment:
Does `Receipt ` and all the fields need to be public? The same for `Block`? Can we make it private or `pup (crate)`?